src/main/java/com/sarality/action/NavigationStyleEnum.java

```diff
@@ -0,0 +1,25 @@
+package com.sarality.action;
+
+import android.content.Intent;
+
+/**
+ * Defines the different types of navigation behaviours that are available to an application at the
+ * time of starting an activity
+ *
+ * @author satya@ (Satya Puniani)
+ */
+public enum NavigationStyleEnum {
+  STANDARD(0),
+  NO_HISTORY(Intent.FLAG_ACTIVITY_NO_HISTORY),
+  KEEP_ON_TOP(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```

Comment:
Is KEEP_ON_TOP same thing as using this activity being called as the latest version? We may need a better name at some point but let's go with this right now and maybe add a TODO here to change name to something more descriptive. Might become easier once we add more navigation styles